### ISSUE TITLE
Changes made on device's calendars are not reflected immediately in the calendar view

### ISSUE THREAD

**Issue by daividssilverio** [toggl/mobileapp#6583]
How to replicate:

1. Setup your calendar integration
2. Have at least your "default calendar" added in the Calendar Settings view
3. Add an event in your "default calendar"
4. Go back to the calendar view
5. You should see the event there
6. Now try editing the event in the "default calendar"
6.1 Go back to calendar view in the toggl app
6.2. You should see that the changes have not been propagated 
7. Try to delete the event in the "default calendar"
7.1. Go back to calendar view in the toggl app
7.2. You should see that the changes have not been propagated and the calendar event stays in place in the toggl calendar view until you restart/try to start a time entry from the deleted calendar event

**Comment by heytherewill** [comment]
I've tried changing the query in our app to something like this 
```
            var uriBuilder = Instances.ContentUri.BuildUpon();
            ContentUris.AppendId(uriBuilder, start.ToUnixTimeMilliseconds());
            ContentUris.AppendId(uriBuilder, end.ToUnixTimeMilliseconds());

            var cursor = appContext.ContentResolver.Query(
                uriBuilder.Build(),
                eventsProjection,
                "deleted != 1",
                null,
                $"{Instances.Begin} ASC");
```

This, [as per docs](https://developer.android.com/guide/topics/providers/calendar-provider#query-instances), would be enough to get all items. There seems to be some kind of cache, however, and the events are almost always up to date, but not always.

I've tried a couple of variations of the above, none of which yielded any results, so I'm going to focus on something else for the time being and let someone else work on this

**Comment by daividssilverio** [comment]
Hey @heytherewill ... so after some testing I noticed a pattern. 

It seems that events from the default "My Calendar" syncs with our app alright with no problems whatsoever.

Now when I try to sync events linked to my e-mail providers (google), the caching problem happens. The api we are using is able to register the new event, but any changes made to it aren't received (even deletion).

**Comment by daividssilverio** [comment]
It gets even more interesting...
I've tried changing the event time from the Samsung default calendar and surprise surprise... it updates the event shown in the app's calendar! Even more interesting, the Google calendar isn't updated immediately after that, it will only notice the change after a few minutes.

Further testing shows that this might be something out of our control...
If I use the "refresh" feature of the google calendar, the events changes are propagated properly to our app.

Also, I've tested with the Samsung account calendar provider, and it works perfectly fine there as well.
Might be worth checking out other providers like Outlook or something like that.

I'd suggest us to close this issue and maybe create another issue to add a note in the calendar settings letting the users know that the Google calendar does not up update immediately, and that they can refresh the google calendar to force it to update. ping @maritoggl ➿ 

Other @toggl/androids might know something else that I'm not aware of, maybe a workaround for this situation.
